### PR TITLE
remove all event listeners when destroying

### PR DESCRIPTION
**Pull request by nightwing**
also adds higher zIndex

### CODE REVIEW

File: code-blast.js [hunk 1]
```diff
@@ -225,33 +225,36 @@ https://twitter.com/JoelBesada/status/670343885655293952
 			canvas.style.position = 'fixed';
 			canvas.style.top = 0;
 			canvas.style.left = 0;
-			canvas.style.zIndex = 1;
+			canvas.style.zIndex = 2147483647;
```
Comment (leMaik) [inline code-blast.js:228]: I can see why this is needed, but maybe you should change this to a more reasonable number? :wink: 
For example, 42 or 9001 (over 9000!) :smile:

**Review by leMaik — CHANGES_REQUESTED** [review]

**Comment by leMaik** [comment]
Thanks for the PR, @nightwing! The changes are well appreciated and I'll merge this soon. :+1:

File: code-blast.js (hunk 1 above)
Comment (nightwing) [reply code-blast.js:228]: that way  canvas can end up behind the editor. This the largest possible value for z-index.

Comment (leMaik) [reply code-blast.js:228]: Fine, but I'm still not convinced that the largest possible value is required here. :thinking:

File: code-blast.js [hunk 2]
```diff
@@ -205,9 +205,9 @@ https://twitter.com/JoelBesada/status/670343885655293952
 		}
 
 		function onCodeMirrorChange(e) {
-			throttledShake(0.3);
-
 			setTimeout(function() {
+				if (!e) e = { start: editor.getCursorPosition() };
```
Comment (leMaik) [inline code-blast.js:209]: Why should `e` ever be undefined?

File: code-blast.js [hunk 3]
```diff
@@ -205,9 +205,9 @@ https://twitter.com/JoelBesada/status/670343885655293952
...
+				throttledShake(0.3);
```
Comment (leMaik) [inline code-blast.js:210]: @Why was this line moved? ~This will delay the shake.~

File: code-blast.js (hunk 1 above)
Comment (nightwing) [reply code-blast.js:228]: if ace is in a popup, which has higher z-index than canvas, canvas will be behind the editor, and will not be visible.

File: code-blast.js (hunk 3 above)
Comment (nightwing) [reply code-blast.js:210]: Generally it si good to avoid dom manipulation from the change event, since it can cause reflow making ace slow.
The shake is not delayed in a visible way.

File: code-blast.js (hunk 2 above)
Comment (nightwing) [reply code-blast.js:209]: I wanted to add a public method to allow triggering shake when the option is activated https://github.com/leMaik/code-blast-ace/pull/4/files#diff-4730e00fd5d64cae591e32dd42da6a10R257
For use case like https://github.com/c9/core/issues/386#issuecomment-294366386

File: code-blast.js (hunk 3 above)
Comment (leMaik) [reply code-blast.js:210]: Okay, indeed the shake is only delayed until the next tick.

File: code-blast.js (hunk 2 above)
Comment (leMaik) [reply code-blast.js:209]: Oh, there's a `blastAway` method below, I didn't see that. Good addition.

**Review by leMaik — APPROVED** [review]
